ledger/error/src/main/scala/com/daml/error/definitions/groups/PartyManagementServiceErrorGroup.scala

```diff
@@ -0,0 +1,152 @@
+    case class Reject(party: String, reason: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends DamlError(
+          cause = s"Update operation for party '$party' failed due to: $reason"
+        ) {
+      override def resources: Seq[(ErrorResource, String)] = Seq(
+        ErrorResource.Party -> party
+      )
+    }
+  }
+
+  @Explanation("""|A party can have only a limited amount of annotations.
+                  |There was an attempt to allocate or update a party with too many annotations.""")
+  @Resolution(
+    "Retry with a smaller number of annotations or delete some of the already existing annotations of this party."
+  )
+  object MaxPartyAnnotationsSizeExceeded
+      extends ErrorCode(
+        id = "MAX_PARTY_DETAILS_ANNOTATIONS_SIZE_EXCEEDED",
+        ErrorCategory.InvalidGivenCurrentSystemStateOther,
+      ) {
+    case class Reject(party: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends DamlError(
+          cause = s"Annotations size for party '$party' has been exceeded"
+        ) {
+      override def resources: Seq[(ErrorResource, String)] = Seq(
+        ErrorResource.Party -> party
+      )
+    }
+  }
+
+  @Explanation(
+    """|Concurrent updates to a party can be controlled by optionally supplying an update request with a resource version.
+                  |A party's resource version can be obtained by reading the party on the Ledger API.
+                  |There was attempt to update a party using a stale resource version indicating a different processes had updated the party earlier."""
+  )
+  @Resolution(
+    """|Restart the party update procedure by reading this party details again to obtain its most recent state and
+                 |in particular its most recent resource version."""
+  )
+  object ConcurrentPartyDetailsUpdateDetected
+      extends ErrorCode(
+        id = "CONCURRENT_PARTY_DETAILS_UPDATE_DETECTED",
+        ErrorCategory.ContentionOnSharedResources,
+      ) {
+    case class Reject(party: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends DamlError(
+          cause =
+            s"Update operation for party '$party' failed due a concurrent update to the same party"
+        ) {
+      override def resources: Seq[(ErrorResource, String)] = Seq(
+        ErrorResource.Party -> party
+      )
+    }
+  }
+
+  @Explanation("The party referred to by the request was not found.")
+  @Resolution(
+    "Check that you are connecting to the right participant node and the party is spelled correctly."
+  )
+  object PartyNotFound
+      extends ErrorCode(
+        id = "PARTY_NOT_FOUND",
+        ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
+      ) {
+    case class Reject(operation: String, party: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends DamlErrorWithDefiniteAnswer(
+          cause = s"$operation failed for unknown party '$party'"
```

Comment:
```suggestion
          cause = s"Party '$party' not found in $operation"
```

Reply:
I prefer the original more. See examples below. Wdyt?

1)
> updating party record failed for unknown party 'abc123'

vs.

> Party 'abc123' not found in updating party record

2)
> creating party record failed for unknown party 'abc123'

vs.

> Party 'abc123' not found in creating party record

Reply:
For now going for:
```
s"Party: '$party' was not found when $operation"
```

E.g.:

> Party: 'abc123' was not found when updating a party record